### PR TITLE
Remove extra space from a Questionnaire View when the group headers have no text to show.

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[issue number]

**Description**
 1. If there is no text to be displayed in `QuestionnaireItemHeaderView` i.e. No prefix, question or hint, then set the visibility of the View as GONE. Otherwise there will be blank space.

>  2. Removed `ErrorView` from `QuestionnaireItemGroupViewHolder` as the group header won't have any errors to show.

 Based on Jing's comment below, error can be shown in the `QuestionnaireItemGroupViewHolder`. So, keeping `ErrorView`.

**Alternative(s) considered**
The alternative would be to weed out such `QuestionnaireItemComponent` while processing `Questionnaire` so that they are not in the list provided to the `RecyclerView`.

**Type**
Choose one: Bug fix 

**Screenshots (if applicable)**
Black lines here are the `DividerItemDecoration` to display the rows in the recyclerview.

**Before:** 
Screenshot_with_empty_group_header_visible

**After:**
Screenshot_with_empty_group_header_gone

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderViewTest.kt [hunk 1]
```diff
@@ -84,8 +84,35 @@ class QuestionnaireItemHeaderViewTest {
...
+            }
+          )
+      }
+    )
 
     assertThat(view.findViewById<TextView>(R.id.hint).visibility).isEqualTo(View.GONE)
   }
+
+  @Test
+  fun shouldHideHeaderView() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderViewTest.kt:103]: add a basic test case to check that we do set the header to visible as well.

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt [hunk 2]
```diff
@@ -17,32 +17,26 @@
...
 
       override fun bind(questionnaireItemViewItem: QuestionnaireItemViewItem) {
         header.bind(questionnaireItemViewItem.questionnaireItem)
       }
 
       override fun displayValidationResult(validationResult: ValidationResult) {
-        error.text =
-          if (validationResult.getSingleStringValidationMessage() == "") null
-          else validationResult.getSingleStringValidationMessage()
+        // No user input
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt]: I have retained the error view and wrapped the header + error view inside a container view which has its visibility tied to header view's visibility. 
For empty headers, the whole row will be gone.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactoryTest.kt [hunk 3]
```diff
@@ -124,4 +86,34 @@ class QuestionnaireItemGroupViewHolderFactoryTest {
       )
       .isEqualTo(View.GONE)
   }
+
+  @Test
+  fun shouldHaveHeaderViewVisible() {
+
```
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactoryTest.kt]: done

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderViewTest.kt (hunk 1 above)
Comment (aditya-07) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderViewTest.kt:121]: Done.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt]: Added error view's testcases as well that were removed in the previous change.

File: datacapture/src/main/res/layout/questionnaire_item_group_header_view.xml [hunk 4]
```diff
@@ -16,19 +16,27 @@
...
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-    />
-
-    <include android:id="@+id/error" layout="@layout/input_error_text_view" />
-
+        android:layout_marginHorizontal="@dimen/item_margin_horizontal"
+        android:layout_marginVertical="@dimen/item_margin_vertical"
+        android:orientation="horizontal"
+        tools:ignore="UselessParent"
```
Comment (jingtang10) [inline datacapture/src/main/res/layout/questionnaire_item_group_header_view.xml:30]: why not just use the top level Linear layout?

Comment (aditya-07) [reply datacapture/src/main/res/layout/questionnaire_item_group_header_view.xml]: Changed it to have the same flat view hierarchy. Now the header and error views can independently be made visible or gone.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt [hunk 5]
```diff
@@ -43,6 +43,8 @@ internal object QuestionnaireItemGroupViewHolderFactory :
         error.text =
           if (validationResult.getSingleStringValidationMessage() == "") null
           else validationResult.getSingleStringValidationMessage()
+
+        error.visibility = if (error.text.isNotEmpty()) View.VISIBLE else View.GONE
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt:47]: Can you use `isEmptyOrNull`?

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt (hunk 5 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemGroupViewHolderFactory.kt:47]: discussed with @aditya-07 offline - text view always returns non-null string, so this should be good.

**Pull request merged**